src/main/kotlin/interpreter/ProgramInterpreter.kt

```diff
@@ -0,0 +1,17 @@
+package org.example.interpreter
+
+import org.example.ast.*
+
+
+class ProgramInterpreter(private val node: ProgramNode): Interpreter {
```

Comment:
Idea for the future: We could add an environment as an optional value here. It would make testing this interpreter more effective. Right now the tests for it just check that it doesn't break.

I understand why you did it this way though, we should think of way to let it be mutable inside the interpreters but read-only from the outside.